include/seastar/core/shared_ptr.hh

```diff
@@ -343,6 +343,7 @@ public:
     lw_shared_ptr& operator=(std::nullptr_t) noexcept {
         return *this = lw_shared_ptr();
     }
+    [[deprecated("Move into the pointee instead")]]
```

Comment:
Moving into the pointee has a different effect - all other sharers will see the new value.

The real replacement is to call make_lw_shared\<> and assign the result.